src/mqueryfront/src/queryUtils.js

```diff
@@ -0,0 +1,38 @@
+export const isStatusFinished = (status) =>
+    ["done", "cancelled", "failed", "expired"].includes(status);
+
+export const getClassForStatus = (status) => {
+    let classSufix = null;
+
+    switch (status) {
+        case "done":
+            classSufix = "success";
+            break;
+        case "processing":
+        case "querying":
+            classSufix = "info";
+            break;
+        case "expired":
+            classSufix = "warning";
+            break;
+        case "cancelled":
+            classSufix = "danger";
+            break;
+        default:
+            break;
```

Comment:
maybe add some console log to make future debugging easier here.

Reply:
OK. Unknown status will be "visible" by lack of colouring. I added: 
console.log(`getClassForStatus: unknown status="${status}"`);